Sources/GravatarUI/SwiftUI/OAuthSession/KeychainToken.swift

```diff
@@ -0,0 +1,30 @@
+import Foundation
+
+struct KeychainToken: Codable {
+    let token: String
+    var isExpired: Bool = false
+
+    init?(data: Data) {
+        let decoder = JSONDecoder()
+        do {
+            let decodedToken = try decoder.decode(KeychainToken.self, from: data)
+            self = decodedToken
+        } catch {
+            return nil
+        }
+    }
+
+    init(token: String) {
+        self.token = token
+    }
+
+    var toData: Data? {
+        let encoder = JSONEncoder()
+        do {
+            let data = try encoder.encode(self)
+            return data
+        } catch {
+            return nil
+        }
```

Comment:
If we only need to return nil, we can do `try?` (same on the init)
```suggestion
        let encoder = JSONEncoder()
        return try? encoder.encode(self)
```